python/tk_framework_alias/server/socketio/api_request.py

```diff
@@ -89,7 +94,19 @@ def execute(self, request_name):
         :type request: str
         """
 
-        raise NotImplementedError("Subclass must implement")
+        self.validate(request_name)
```

Comment:
If this is an abstract base class, why do we need to have an implementation here?

Reply:
A base abstract class can still have function implementations; it is abstract though because it has some methods that must be implemented by the subclass, making this base class not directly instantiable.

We include this function implementation for this function because we want all subclasses to use this behavior, and avoids duplicating the code in each subclass.